code/game/machinery/keyfab.dm

```diff
@@ -0,0 +1,86 @@
+/obj/machinery/keyfab
+	name = "key fabricator"
+	desc = "A machine that can print keys"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "old_recharger"
+	density = 0
+	anchored = 1
+	var/busy = 0
+	var/obj/item/key/storedkey
+	var/build_time = 20 SECONDS
+
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE
+
+/obj/machinery/keyfab/New()
+	..()
+	component_parts = newlist(
+							/obj/item/weapon/circuitboard/keyfab,
+							/obj/item/weapon/stock_parts/micro_laser
+							)
+	RefreshParts()
+	update_icon()
+
+/obj/machinery/keyfab/RefreshParts()
+	var/T = 0
+	for(var/obj/item/weapon/stock_parts/micro_laser/ML in component_parts)
+		T = ML.rating
```

Comment:
I realize there's only one, but this would be very unexpected if someone wanted to increase it to 2 microlasers or something later on.

Reply:
Funny you should say that. I forget where or when it was but I was told the exact opposite in the past. Since it was only 1 part you should do = rather than +=. Not a huge issue so I will change it.

Reply:
Who told you that? I'll see what their reasoning is

Reply:
God himself I guess? I don't recall when or where I heard it.

Reply:
Alright, it's not that important. If you remember, let me know and we'll find out why they think they way they do. As Dr. Mobius says, it might be worth a cognitive realignment if the theory is better.